src/storage/src/source/reclock.rs

```diff
@@ -964,518 +676,457 @@ mod tests {
+        ];
+        let source_upper = partitioned_frontier([(PART_ID, MzOffset::from(5))]);
 
-        now_fn.advance(Duration::from_millis(1000));
-        follower.push_trace_batch(operator.mint(&source_upper).await);
+        follower.push_trace_batch(operator.mint(source_upper.borrow()).await);
         let reclocked_msgs = follower
-            .reclock(&mut batch)
-            .expect("beyond source frontier")
-            .expect("we should have all required bindings")
-            .consume_all();
-        assert_eq!(reclocked_msgs, &[(3, 1000.into()), (4, 1000.into())]);
-        assert!(batch[&PART_ID].is_empty());
+            .reclock(batch)
+            .map(|(m, ts)| (m, ts.unwrap()))
+            .collect_vec();
+        assert_eq!(reclocked_msgs, &[(3, 2000.into()), (4, 2000.into())]);
 
         // Reclock the same offsets again
-        batch.insert(
-            PART_ID,
-            vec![(1, MzOffset::from(1)), (2, MzOffset::from(2))],
-        );
+        let batch = vec![
+            (1, Partitioned::with_partition(PART_ID, MzOffset::from(1))),
+            (2, Partitioned::with_partition(PART_ID, MzOffset::from(2))),
+        ];
+
         let reclocked_msgs = follower
-            .reclock(&mut batch)
-            .expect("beyond source frontier")
-            .expect("we should have all required bindings")
-            .consume_all();
-        assert_eq!(reclocked_msgs, &[(1, 0.into()), (2, 0.into())]);
-        assert!(batch[&PART_ID].is_empty());
+            .reclock(batch)
+            .map(|(m, ts)| (m, ts.unwrap()))
+            .collect_vec();
+        assert_eq!(reclocked_msgs, &[(1, 1000.into()), (2, 1000.into())]);
 
         // Reclock a batch with offsets that spans multiple bindings
-        batch.insert(
-            PART_ID,
-            vec![
-                (1, MzOffset::from(1)),
-                (2, MzOffset::from(2)),
-                (3, MzOffset::from(3)),
-                (4, MzOffset::from(4)),
-            ],
-        );
+        let batch = vec![
+            (1, Partitioned::with_partition(PART_ID, MzOffset::from(1))),
+            (2, Partitioned::with_partition(PART_ID, MzOffset::from(2))),
+            (3, Partitioned::with_partition(PART_ID, MzOffset::from(3))),
+            (4, Partitioned::with_partition(PART_ID, MzOffset::from(4))),
+        ];
         let reclocked_msgs = follower
-            .reclock(&mut batch)
-            .expect("beyond source frontier")
-            .expect("we should have all required bindings")
-            .consume_all();
+            .reclock(batch)
+            .map(|(m, ts)| (m, ts.unwrap()))
+            .collect_vec();
         assert_eq!(
             reclocked_msgs,
             &[
-                (1, 0.into()),
-                (2, 0.into()),
-                (3, 1000.into()),
-                (4, 1000.into())
+                (1, 1000.into()),
+                (2, 1000.into()),
+                (3, 2000.into()),
+                (4, 2000.into()),
```

Comment:
The timestamps in this test have changed only because the reclock operator always waits for one tick on construction and not because the interpretation of the remap trace changed.

Reply:
Yeah we have a lot of reclock tests with various combinations